### PR TITLE
[app-store-promote] Migrated to Node 10

### PR DESCRIPTION
**Pull request by max-zaytsev**
**Task name**: app-store-promote

**Description**: Migrated task to Node 10.

**Documentation changes required:** No

**Merge after this PR is complete:**  https://github.com/microsoft/build-task-team/issues/676

**Added unit tests:** No

**Attached related issue:** [#668](https://github.com/microsoft/build-task-team/issues/668)

**Hotfix branch:** [node6hotfixes/app-store-promote](https://github.com/microsoft/app-store-vsts-extension/pull/new/node6hotfixes/app-store-promote)

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/app-store-promote/package.json [hunk 1]
```diff
@@ -4,9 +4,14 @@
   "description": "Azure Pipelines task to promote a build in the Apple App Store",
   "main": "app-store-promote.js",
   "dependencies": {
-    "azure-pipelines-task-lib": "^2.11.1"
+    "azure-pipelines-task-lib": "^3.0.6-preview.0",
+    "@types/node": "^10.17.0",
+    "@types/mocha": "^5.2.7",
+    "@types/uuid": "^8.3.0"
```
Comment (DaniilShmelev) [inline Tasks/app-store-promote/package.json:10]: nit: do we need types for uuid here? I don't see this package being used anywhere

**Comment by DaniilShmelev** [comment]
Also, could you please update deprecated MochaDone types in tests?

**Comment by max-zaytsev** [comment]
> Also, could you please update deprecated MochaDone types in tests?
Thanks! Done)

**Review by DaniilShmelev — APPROVED** [review]
Thanks, LGTM!

**Review by anatolybolshakov — APPROVED** [review]

**Comment by max-zaytsev** [comment]
`gulp packageprod` is failing in CI. Investigating this

**Comment by DaniilShmelev** [comment]
Tested these changes manually. Successful pipeline: https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2441&view=results

**Pull request merged**